src/utils/tests/solidity-keccak256.test.ts

```diff
@@ -0,0 +1,136 @@
+  //     expect(solidityKeccak256(input.types, input.values)).toBe(
+  //       utils.solidityKeccak256(input.types, input.values),
+  //     );
+  //   });
+  // });
+  it('should match ethers.js addresses', () => {
+    const inputs = [
+      {
+        types: ['address'],
+        values: ['0x4d7F1790644Af787933c9fF0e2cff9a9B4299Abb'],
+      },
+      {
+        types: ['address', 'address'],
+        values: [
+          '0xB5503a7db1A9105cd459D99153e69a76a8EF1530',
+          '0xaa0fc255b079e775f9307e5cfec472a555cebc3a',
+        ],
+      },
+    ];
+    testSolidityKeccak256(inputs);
+  });
+  it('should match ethers.js strings', () => {
+    const inputs = [
+      { types: ['string'], values: ['essential-eth'] },
+      { types: ['string', 'string'], values: ['firstText', 'secondString'] },
+      {
+        types: ['string', 'string', 'string'],
+        values: ['example1', '2934823', 'true'],
+      },
+    ];
+    testSolidityKeccak256(inputs);
+  });
+  it('should match ethers.js bytes (dynamic size) & BytesLike', () => {
+    const inputs = [
+      { types: ['bytes'], values: [[115, 101, 99, 114, 101, 116]] },
+      { types: ['bytes'], values: ['0x62797465734c696b65'] }, // BytesLike string
+      {
+        types: ['bytes', 'bytes'],
+        values: [
+          '0x657373656e7469616c2d657468',
+          [115, 109, 97, 108, 108, 101, 115, 116],
+        ],
+      },
+    ];
+    testSolidityKeccak256(inputs);
+  });
+  it('should match ethers.js bytes (static size)', () => {
+    const inputs = [
+      { types: ['bytes4'], values: [[116, 101, 115, 116]] },
+      {
+        types: ['bytes13', 'bytes1', 'bytes2', 'bytes5', 'bytes1'],
+        values: [
+          [101, 115, 115, 101, 110, 116, 105, 97, 108, 45, 101, 116, 104],
+          [32],
+          [105, 115],
+          [103, 114, 101, 97, 116],
+          [33],
+        ],
+      },
+    ];
+    testSolidityKeccak256(inputs);
+  });
+  it('should match ethers.js booleans (specified as type "bool" to match solidity)', () => {
+    const inputs = [
+      { types: ['bool'], values: [true] },
+      { types: ['bool', 'bool'], values: [false, false] },
+    ];
+    testSolidityKeccak256(inputs);
+  });
+  it('should match ethers.js (signed and unsigned) integers', () => {
+    const inputs = [
+      { types: ['int16'], values: [-1] },
+      { types: ['uint48'], values: [12] },
+      { types: ['int16', 'uint48'], values: [-1, 12] },
+    ];
+    testSolidityKeccak256(inputs);
+  });
+  it('should match ethers.js array parsing', () => {
+    const inputs = [
+      { types: ['string[2]'], values: [['some-text', 'additional-text']] },
+      {
+        types: ['uint48[3]', 'bool[2]', 'address[2]', 'bytes1[1]'],
+        values: [
+          [2, 5, 8],
+          [true, false],
+          [
+            '0xB5503a7db1A9105cd459D99153e69a76a8EF1530',
+            '0xaa0fc255b079e775f9307e5cfec472a555cebc3a',
+          ],
+          [[15]],
```

Comment:
@dawsbot this input should be valid in theory but fails in both `essential-eth` and `ethers` because `essential-eth` relies on code I believe to be broken that comes from `ethers`. Parsing in an array triggers the `_pack` fn to call itself passing the `isArray` parameter as true (https://github.com/Earnifi/essential-eth/blob/675424d214951ba0dcf3be071cf5c6e641bd2578/src/utils/solidity-keccak256.ts#L85), which triggers `_pack` to add zeros stored in the `Zeros` const to the original `15` value passed in (https://github.com/Earnifi/essential-eth/blob/675424d214951ba0dcf3be071cf5c6e641bd2578/src/utils/solidity-keccak256.ts#L67). This then triggers `arrayify` to throw an error because the number passed in is too large (https://github.com/Earnifi/essential-eth/blob/675424d214951ba0dcf3be071cf5c6e641bd2578/src/utils/bytes.ts#L101). Any thoughts on remedying this? We could aim to fix this functionality in `essential-eth`, but then it won't match `ethers`. From what I can tell, this seems like a bug in `ethers` opposed to expected functionality